client/src/axiom/axiom.ts

```diff
@@ -113,4 +143,35 @@ export class Axiom<T> {
       throw new Error(e);
     }
   }
+
+  private async prepareSendQuery(ipfsClient?: IpfsClient) {
+    if (this.config.privateKey === undefined) {
```

Comment:
we should change this to this.config.wallet and use the ethers wallet type or something. Private key should only be in constructor to make the function more flexible.